### DISCUSSION TITLE
follow-ups 2 for community-containers

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#3889] [Ideas] [answered]
Follow-ups for https://github.com/nextcloud/all-in-one/discussions/3432
### earlier
- [x] think about instance restore: should teh backup container run twice in a row to restore the setting first and then start the correct restore. Same for normal restore? https://github.com/nextcloud/all-in-one/pull/4707
    - [x] remove the beta hint in the docs after this is resolved
- [x] ~~allow to define nextcloud_variables~~ Not needed anymore due to https://github.com/nextcloud/all-in-one/pull/6568
- [x] ~~allow to define nextcloud_secrets~~ Not needed anymore due to https://github.com/nextcloud/all-in-one/pull/6568
    - [x] ~~adjust https://github.com/nextcloud/all-in-one/blob/5cbbe1bab8f5e09af22b1d93ed1d79e662c4b023/community-containers/facerecognition/facerecognition.json#L13 if both are available~~ Done in https://github.com/nextcloud/all-in-one/pull/6835
- [x] ~~instead of modifying container to put a caddy import into place, should there be an option to modify a subpath that is automatically forwarded to the correct container in a subpath available on the nextcloud domain? https://github.com/nextcloud/all-in-one/pull/3192 (maybe also use that in the containers but must also work without)~~
    - [x] ~~move onlyoffice to community apps afterwards? (Also remove all onlyoffice logic from aio then) -- needs testing with jwt secret - maxbe sheet with the secret for now...~~

### later
- [x] add containers
	- [x] add backup container that allows to view the backup, restore files, etc. https://github.com/nextcloud/all-in-one/pull/5889
	- [x] smbserver https://github.com/nextcloud/all-in-one/pull/6162
- [x] add on the fly notification? Maybe get notificcations via nc into the nc container and then use the admin notify scirpt... tracked now in https://github.com/nextcloud/all-in-one/discussions/6836
    - [x] ~~add off-shore backup container afterwards for regular backups -> needs to be able to notify of failed or succesful backup~~ Not a good idea, because it would be stopped by the daily backup feature... So abandoned this idea for now.
    - [x] https://github.com/szaimen/aio-scrutiny/issues/3
- [x] https://github.com/nextcloud/all-in-one/issues/5938
- [x] ~~allow to set not defined environmental variables from the aio interface: by providing an input field and regex and maybe even description).~~

**Comment by luzfcb** [comment]
It should be good to have an option to pass additional arguments to the `docker run` command per container. (related to https://github.com/nextcloud/all-in-one/discussions/4277) 

https://docs.docker.com/config/containers/resource_constraints/#gpu
https://docs.docker.com/engine/alternative-runtimes/

**Reply by szaimen** [reply]
Someone started working on that in https://github.com/nextcloud/all-in-one/pull/5132 but you saw that already

**Reply by szaimen** [reply]
This is now implemented since a while

**Comment by szaimen** [comment]
I've added this to the list in https://github.com/nextcloud/all-in-one/issues/5251 now

**Comment by szaimen** [comment, marked as answer]
This is mostly done. Tracking remaining task in https://github.com/nextcloud/all-in-one/discussions/6836
